<center>
    <img src="https://media-exp1.licdn.com/dms/image/C4E1BAQErMN84hgIX3Q/company-background_10000/0/1622546957980?e=2147483647&v=beta&t=l7tU8QXWhNOtm4NZ3ZLXRSw42o0U-QZu-3JbvBr94Fc" width="1000" alt="cognitiveclass.ai logo"  />
</center>

<h1 style="text-align:center; font-size: 5em;">Working with APIs</h1>

## Working with APIS
<div class="alert alert-block alert-danger">
    <b>Number one key when working with APIs is to read the documentation. APIs will have different parameters and structure, depending on how it was created.</b>
</div>

<div class="alert alert-block alert-warning">
    <b>Task:</b> create a <code>config.py</code> file to hold your API keys.
    <li> Reason: API Keys can cost money per number of calls. If we expose our keys on the code it can be used by unknown parties. 
</div>

`python -m pip install requests`

In [1]:
# Dependencies
import requests
from pprint import pprint
import json
import pandas as pd

In [ ]:
import sys
import os

# Add the parent directory to sys.path
#parent_dir = os.path.abspath(os.path.join(os.getcwd(), '..'))
#sys.path.insert(0, parent_dir)

# Now you can import config.py
#from config import movies_api_key

# Use the config values
#print(movies_api_key)

<a name="anchorQueryBasics" style="position:absolute;"></a>
<hr style="border:2px solid">

# Query Basics
<hr style="border-top:1px dashed">

[Requests Documentation](https://docs.python-requests.org/en/latest/)

[Movie Info API](http://www.omdbapi.com/)

In [3]:
##Validate your API key:
movies_api_key = 'c3ba08dc'
api_key = movies_api_key

In [4]:
##Movie Search
##Setting the parameters for the search
url = 'http://www.omdbapi.com/?t=' # ?t= refers to the title of the movie
movie = 'Casino Royale' ## Change to any movie of your preference
#movie = 'UP'
api_key_str = "&apikey=" + api_key

print('call to send to the API:', url + movie + api_key_str)

call to send to the API: http://www.omdbapi.com/?t=Casino Royale&apikey=c3ba08dc


In [5]:
response = requests.get(url + movie + api_key_str)
response
##If you get "<Response [200]>" then everything is fine, otherwise review your code, specially your API key.

<Response [200]>

In [6]:
##Print the queried url. NOTE: Make sure to comment out as it will also print the API key
print(response.url)

http://www.omdbapi.com/?t=Casino%20Royale&apikey=c3ba08dc


#### URL validation:

You can also validate your response by copying the URL from above and past it to a web browser.

__Is it similar to any the Python datatypes that you know?__ 


In [7]:
# Check status code. A 200 means everything is ok with the syntax of the query.
response.status_code

200

**2XX Codes**<br>
<div class="alert alert-block alert-warning">
   These codes convey that the server has received the client’s request and processed it successfully. Here are the most common examples:
    <br>
</div>

**200 OK**: The request was successful.<br>
**201 Created**: The resource has been created on the server. This response is typically returned for POST requests.<br>
**202 Accepted**: The request has been received but is still being processed.<br>
**204 No Content**: The request has been successfully processed but no content will be returned.

'utf-8'

## Data analysis

In [9]:
##Looking at the different functions and attribute of "response"
help(response)

Help on Response in module requests.models object:

class Response(builtins.object)
 |  The :class:`Response <Response>` object, which contains a
 |  server's response to an HTTP request.
 |  
 |  Methods defined here:
 |  
 |  __bool__(self)
 |      Returns True if :attr:`status_code` is less than 400.
 |      
 |      This attribute checks if the status code of the response is between
 |      400 and 600 to see if there was a client error or a server error. If
 |      the status code, is between 200 and 400, this will return True. This
 |      is **not** a check to see if the response code is ``200 OK``.
 |  
 |  __enter__(self)
 |  
 |  __exit__(self, *args)
 |  
 |  __getstate__(self)
 |  
 |  __init__(self)
 |      Initialize self.  See help(type(self)) for accurate signature.
 |  
 |  __iter__(self)
 |      Allows you to use a response as an iterator.
 |  
 |  __nonzero__(self)
 |      Returns True if :attr:`status_code` is less than 400.
 |      
 |      This attribute checks if

In [11]:
##Accessing the content of the response:
print(response.content)

b'{"Title":"Casino Royale","Year":"2006","Rated":"PG-13","Released":"17 Nov 2006","Runtime":"144 min","Genre":"Action, Adventure, Thriller","Director":"Martin Campbell","Writer":"Neal Purvis, Robert Wade, Paul Haggis","Actors":"Daniel Craig, Eva Green, Judi Dench","Plot":"After earning a licence to kill, secret agent James Bond sets out on his first mission as 007. Bond must defeat a private banker funding terrorists in a high-stakes game of poker at Casino Royale, in Montenegro.","Language":"English, Serbian, German, Italian, French","Country":"United States, United Kingdom, Czech Republic, Germany, Bahamas, Italy","Awards":"Won 1 BAFTA Award28 wins & 44 nominations total","Poster":"https://m.media-amazon.com/images/M/MV5BMWQ1ZDM4NDktMWY0NC00MjcxLWJlMDMtNmE2MGVhYzRjMWQ0XkEyXkFqcGc@._V1_SX300.jpg","Ratings":[{"Source":"Internet Movie Database","Value":"8.0/10"},{"Source":"Rotten Tomatoes","Value":"94%"},{"Source":"Metacritic","Value":"80/100"}],"Metascore":"80","imdbRating":"8.0","imdb

In [12]:
# Server's information for this request
response.headers

{'Date': 'Fri, 07 Nov 2025 10:04:48 GMT', 'Content-Type': 'application/json; charset=utf-8', 'Transfer-Encoding': 'chunked', 'Connection': 'keep-alive', 'Cache-Control': 'public, max-age=3600', 'Expires': 'Fri, 07 Nov 2025 11:04:48 GMT', 'Last-Modified': 'Fri, 07 Nov 2025 10:04:48 GMT', 'Vary': '*', 'Server': 'cloudflare', 'X-AspNet-Version': '4.0.30319', 'Access-Control-Allow-Origin': '*', 'cf-cache-status': 'DYNAMIC', 'Content-Encoding': 'gzip', 'CF-RAY': '99ac0230b81086bf-LHR'}

In [13]:
# Convert the response to JSON format
data = response.json()
print(data)

{'Title': 'Casino Royale', 'Year': '2006', 'Rated': 'PG-13', 'Released': '17 Nov 2006', 'Runtime': '144 min', 'Genre': 'Action, Adventure, Thriller', 'Director': 'Martin Campbell', 'Writer': 'Neal Purvis, Robert Wade, Paul Haggis', 'Actors': 'Daniel Craig, Eva Green, Judi Dench', 'Plot': 'After earning a licence to kill, secret agent James Bond sets out on his first mission as 007. Bond must defeat a private banker funding terrorists in a high-stakes game of poker at Casino Royale, in Montenegro.', 'Language': 'English, Serbian, German, Italian, French', 'Country': 'United States, United Kingdom, Czech Republic, Germany, Bahamas, Italy', 'Awards': 'Won 1 BAFTA Award28 wins & 44 nominations total', 'Poster': 'https://m.media-amazon.com/images/M/MV5BMWQ1ZDM4NDktMWY0NC00MjcxLWJlMDMtNmE2MGVhYzRjMWQ0XkEyXkFqcGc@._V1_SX300.jpg', 'Ratings': [{'Source': 'Internet Movie Database', 'Value': '8.0/10'}, {'Source': 'Rotten Tomatoes', 'Value': '94%'}, {'Source': 'Metacritic', 'Value': '80/100'}], 'M

In [14]:
type(data)

dict

In [16]:
data.keys()

dict_keys(['Title', 'Year', 'Rated', 'Released', 'Runtime', 'Genre', 'Director', 'Writer', 'Actors', 'Plot', 'Language', 'Country', 'Awards', 'Poster', 'Ratings', 'Metascore', 'imdbRating', 'imdbVotes', 'imdbID', 'Type', 'DVD', 'BoxOffice', 'Production', 'Website', 'Response'])

In [19]:
data['Actors']

'Daniel Craig, Eva Green, Judi Dench'

In [26]:
##Pprint helps us to visualise better a JSON/dictionary data
data['Ratings'], type(data['Ratings'])
data['Ratings'][2]

{'Source': 'Metacritic', 'Value': '80/100'}

In [ ]:
# Retriveve all the Keys


dict_keys(['Title', 'Year', 'Rated', 'Released', 'Runtime', 'Genre', 'Director', 'Writer', 'Actors', 'Plot', 'Language', 'Country', 'Awards', 'Poster', 'Ratings', 'Metascore', 'imdbRating', 'imdbVotes', 'imdbID', 'Type', 'DVD', 'BoxOffice', 'Production', 'Website', 'Response'])

In [ ]:
# Retriveve all Values


dict_values(['Casino Royale', '2006', 'PG-13', '17 Nov 2006', '144 min', 'Action, Adventure, Thriller', 'Martin Campbell', 'Neal Purvis, Robert Wade, Paul Haggis', 'Daniel Craig, Eva Green, Judi Dench', 'After earning a licence to kill, secret agent James Bond sets out on his first mission as 007. Bond must defeat a private banker funding terrorists in a high-stakes game of poker at Casino Royale, in Montenegro.', 'English, Serbian, German, Italian, French', 'United States, United Kingdom, Czech Republic, Germany, Bahamas, Italy', 'Won 1 BAFTA Award28 wins & 44 nominations total', 'https://m.media-amazon.com/images/M/MV5BMWQ1ZDM4NDktMWY0NC00MjcxLWJlMDMtNmE2MGVhYzRjMWQ0XkEyXkFqcGc@._V1_SX300.jpg', [{'Source': 'Internet Movie Database', 'Value': '8.0/10'}, {'Source': 'Rotten Tomatoes', 'Value': '94%'}, {'Source': 'Metacritic', 'Value': '80/100'}], '80', '8.0', '726,077', 'tt0381061', 'movie', 'N/A', '$167,445,960', 'N/A', 'N/A', 'True'])

### Navigating throught the "response"

Because the data is now a dictionary, it might the helpful to review the session where we saw this subject. 

In [ ]:
# Get individual Responses from the JSON file


Movie director Martin Campbell
Amount made at the box office $167,445,960


In [45]:
# From multiple values
## Lets analyse the "rating" of the movie
####First level:
type(data['Ratings'])
len(data['Ratings'])
print(data['Ratings'])
data['Ratings'][0], type(data['Ratings'][0])
data['Ratings'][0].keys()
data['Ratings'][0]['Source'], data['Ratings'][0]['Value']
print('platform :', data['Ratings'][0]['Source'])
print('raiting given:', data['Ratings'][0]['Value'])
print('platform :', data['Ratings'][1]['Source'])
print('raiting given:', data['Ratings'][1]['Value'])
print('platform :', data['Ratings'][2]['Source'])
print('raiting given:', data['Ratings'][2]['Value'])

[{'Source': 'Internet Movie Database', 'Value': '8.0/10'}, {'Source': 'Rotten Tomatoes', 'Value': '94%'}, {'Source': 'Metacritic', 'Value': '80/100'}]
platform : Internet Movie Database
raiting given: 8.0/10
platform : Rotten Tomatoes
raiting given: 94%
platform : Metacritic
raiting given: 80/100


In [ ]:
# From multiple values

##Accessing the first element of the list:


<class 'dict'>
{'Source': 'Internet Movie Database', 'Value': '8.0/10'}


In [ ]:
##Accessing the key, value pair:

##Raiting source

##Raiting:


dict_keys(['Source', 'Value'])
Internet Movie Database
8.0/10


__Exercise:__ Print all the raitings available in "response"

In [ ]:
##Your code:


Internet Movie Database: 8.0/10
Rotten Tomatoes: 94%
Metacritic: 80/100


__Exercise:__ Create a function that, given an user input, returns the following data (e.g. movie UP): 

In [ ]:
## Your code here: 
def movie_function(movie: str):
    """
        Funcition to get some movie facts.
        Parameters:
            movie (str): name of your movie
    """
    movies_api_key = 'd998895c'
    api_key = movies_api_key
    url = 'http://www.omdbapi.com/?t=' # ?t= refers to the title of the movie
    api_key_str = "&apikey=" + api_key
    response = requests.get(url + movie + api_key_str)
    data = response.json()
    director = data['Director']
    box_office = data['BoxOffice']
    ratings = data['Ratings']
    print("Movie Title:", movie)
    print("Director:", director)
    print("Box Office:", box_office)
    for source in range(len(ratings)):
        print(ratings[source]['Source'], ": ", ratings[source]['Value'])
 
movie_function('Home Alone')
#movie_function('Up')
 


Movie Title: Home Alone
Director: Chris Columbus
Box Office: $285,761,243
Internet Movie Database :  7.7/10
Rotten Tomatoes :  66%
Metacritic :  63/100


<a name="anchorBadQuery" style="position:absolute;"></a>
<hr style="border:2px solid">

# Bad Query
<hr style="border-top:1px dashed">

In [22]:
# Look for all movies
url2 = 'http://www.omdbapi.com/?'
bad_key = '1234'
api = 'apikey='
year = "&y="+ '2019'


In [23]:
## Bad API KEY
#query= url2 + api+ bad_key + year
# Not Formatted properly
query= url2 + api_key + year

In [24]:
print(query)

http://www.omdbapi.com/?c3ba08dc&y=2019


In [25]:
# Performing a GET request 
response2 = requests.get(query)
print(response2)
# Response 400 means it is a bad request

<Response [401]>


In [26]:
# Convert the response to JSON format
action_data = response2.json()
pprint(action_data)

{'Error': 'No API key provided.', 'Response': 'False'}


In [ ]:
##Movie Search
##Setting the parameters for the search
# Look for all movies


'http://www.omdbapi.com/?c3ba08dc&y=2019'

The documentation clearly says that one of the parameters i or t are required for a query.

**4XX Codes**<br>
<div class="alert alert-block alert-warning">
   These codes convey that the client’s request has an error. It’s possible that the request is written incorrectly, or the resource which the client is requesting doesn't exist.
    <br>
</div>

**400 Bad Request**: There is something wrong in the client’s request.<br>
**401 Unauthorized**: The client is not authorized to make this request.<br>
**403 Forbidden**: The request is correct but can’t be processed. Likely, the problem is that the client does not have required permissions.<br>
**404 Not Found**: The resource being requested doesn't exist.

<a name="anchorPag" style="position:absolute;"></a>
<hr style="border:2px solid">

# Pagination
<hr style="border-top:1px dashed">

<div class="alert alert-block alert-warning">
    <b>What does it do?:</b> Pagination limits the number of results to help keep network traffic in check.
    <br>We should read the API's documentation to understand how many values we have per page. 
</div>

Source: https://datahelpdesk.worldbank.org/knowledgebase/articles/898590-country-api-queries

In [47]:
url3 = "http://api.worldbank.org/v2/"
api_format = "json"
query2= f"{url3}countries?format={api_format}"
query2


'http://api.worldbank.org/v2/countries?format=json'

In [48]:
###Validating the reply:
requests.get(query2)

<Response [200]>

In [49]:
# Get country information in JSON format
countries_response = requests.get(query2)

In [52]:
countries_response = countries_response.json()
countries_response

[{'page': 1, 'pages': 6, 'per_page': '50', 'total': 296},
 [{'id': 'ABW',
   'iso2Code': 'AW',
   'name': 'Aruba',
   'region': {'id': 'LCN',
    'iso2code': 'ZJ',
    'value': 'Latin America & Caribbean '},
   'adminregion': {'id': '', 'iso2code': '', 'value': ''},
   'incomeLevel': {'id': 'HIC', 'iso2code': 'XD', 'value': 'High income'},
   'lendingType': {'id': 'LNX', 'iso2code': 'XX', 'value': 'Not classified'},
   'capitalCity': 'Oranjestad',
   'longitude': '-70.0167',
   'latitude': '12.5167'},
  {'id': 'AFE',
   'iso2Code': 'ZH',
   'name': 'Africa Eastern and Southern',
   'region': {'id': 'NA', 'iso2code': 'NA', 'value': 'Aggregates'},
   'adminregion': {'id': '', 'iso2code': '', 'value': ''},
   'incomeLevel': {'id': 'NA', 'iso2code': 'NA', 'value': 'Aggregates'},
   'lendingType': {'id': '', 'iso2code': '', 'value': 'Aggregates'},
   'capitalCity': '',
   'longitude': '',
   'latitude': ''},
  {'id': 'AFG',
   'iso2Code': 'AF',
   'name': 'Afghanistan',
   'region': {'id': 

In [ ]:
len(countries_response)

2

In [58]:
print(countries_response[0])
countries_response[1]
print(len(countries_response[1]))

{'page': 1, 'pages': 6, 'per_page': '50', 'total': 296}
50


In [ ]:
# First element is general information, second is countries themselves
page = 1
query2= f"{url3}countries?&page={page}&format=json"
query2
print(requests.get(query2))
request_pgs =requests.get(query2).json()
request_pgs

<Response [200]>


[{'page': 7, 'pages': 6, 'per_page': '50', 'total': 296}, []]

In [79]:
page = 1
query2= f"{url3}countries?&page={page}&format=json"
query2
print(requests.get(query2))
request_pgs =requests.get(query2).json()
request_pgs

<Response [200]>


[{'page': 1, 'pages': 6, 'per_page': '50', 'total': 296},
 [{'id': 'ABW',
   'iso2Code': 'AW',
   'name': 'Aruba',
   'region': {'id': 'LCN',
    'iso2code': 'ZJ',
    'value': 'Latin America & Caribbean '},
   'adminregion': {'id': '', 'iso2code': '', 'value': ''},
   'incomeLevel': {'id': 'HIC', 'iso2code': 'XD', 'value': 'High income'},
   'lendingType': {'id': 'LNX', 'iso2code': 'XX', 'value': 'Not classified'},
   'capitalCity': 'Oranjestad',
   'longitude': '-70.0167',
   'latitude': '12.5167'},
  {'id': 'AFE',
   'iso2Code': 'ZH',
   'name': 'Africa Eastern and Southern',
   'region': {'id': 'NA', 'iso2code': 'NA', 'value': 'Aggregates'},
   'adminregion': {'id': '', 'iso2code': '', 'value': ''},
   'incomeLevel': {'id': 'NA', 'iso2code': 'NA', 'value': 'Aggregates'},
   'lendingType': {'id': '', 'iso2code': '', 'value': 'Aggregates'},
   'capitalCity': '',
   'longitude': '',
   'latitude': ''},
  {'id': 'AFG',
   'iso2Code': 'AF',
   'name': 'Afghanistan',
   'region': {'id': 

In [82]:
request_pgs[0]['pages']

6

In [68]:
request_pgs =requests.get(query2).json()
request_pgs

[{'page': 1, 'pages': 6, 'per_page': '50', 'total': 296},
 [{'id': 'ABW',
   'iso2Code': 'AW',
   'name': 'Aruba',
   'region': {'id': 'LCN',
    'iso2code': 'ZJ',
    'value': 'Latin America & Caribbean '},
   'adminregion': {'id': '', 'iso2code': '', 'value': ''},
   'incomeLevel': {'id': 'HIC', 'iso2code': 'XD', 'value': 'High income'},
   'lendingType': {'id': 'LNX', 'iso2code': 'XX', 'value': 'Not classified'},
   'capitalCity': 'Oranjestad',
   'longitude': '-70.0167',
   'latitude': '12.5167'},
  {'id': 'AFE',
   'iso2Code': 'ZH',
   'name': 'Africa Eastern and Southern',
   'region': {'id': 'NA', 'iso2code': 'NA', 'value': 'Aggregates'},
   'adminregion': {'id': '', 'iso2code': '', 'value': ''},
   'incomeLevel': {'id': 'NA', 'iso2code': 'NA', 'value': 'Aggregates'},
   'lendingType': {'id': '', 'iso2code': '', 'value': 'Aggregates'},
   'capitalCity': '',
   'longitude': '',
   'latitude': ''},
  {'id': 'AFG',
   'iso2Code': 'AF',
   'name': 'Afghanistan',
   'region': {'id': 

In [62]:
query2= f"{url3}countries?&page={2}&format=json"
request_pgs = requests.get(query2)
##We can see that the request was successful:
request_pgs

<Response [502]>

In [ ]:
https://search.worldbank.org/api/v3/wds?format=json&rows=20&os=0&fl=docty

In [ ]:
# Check number of responses


Number of responses 50


In [ ]:
##We can see that there are more elements to extract:
## 'page': 1, 'pages': 6, 'per_page': '50


{'page': 1, 'pages': 6, 'per_page': '50', 'total': 296}

In [ ]:
##Requesting the second page:


'http://api.worldbank.org/v2/countries?&page=1&format=json'

In [ ]:
##Testing the amount of pages that are in the data source
##Lets see if there is a page = 20



<Response [200]>

In [ ]:
##But accessing the actual data we can see that has no countries info.


{'page': 20, 'pages': 6, 'per_page': '50', 'total': 296}


[]

[{'page': 20, 'pages': 6, 'per_page': '50', 'total': 296}, []]

__Exercise:__ Taking in consideration the previous points, develop a code that will collect all the coutries information from the data source. Do not hardcode the solution, it has to be dynamic.

In [84]:
##Your code here:

##Your code here:

# Create an empty list to append our responses
multiresponse_json=[]

# Create a loop that will give back the response for all pages, the response 
counter = 1
while True: 
    query2= f"{url3}countries?&page={counter}&format=json"
    # Print the different pages for the query
    print(query2)
    response = requests.get(query2).json()
    if len(response[1]) != 0: 
        for county in response[1]:
            multiresponse_json.append(county)
        #multiresponse = response[1]
        counter += 1
    else:
        break

# Output
print("Items extracted: ", len(multiresponse_json))
print('Data sample:')
#pprint(multiresponse_json[0:3])

http://api.worldbank.org/v2/countries?&page=1&format=json
http://api.worldbank.org/v2/countries?&page=2&format=json
http://api.worldbank.org/v2/countries?&page=3&format=json
http://api.worldbank.org/v2/countries?&page=4&format=json
http://api.worldbank.org/v2/countries?&page=5&format=json
http://api.worldbank.org/v2/countries?&page=6&format=json
http://api.worldbank.org/v2/countries?&page=7&format=json
Items extracted:  296
Data sample:


In [87]:
len(multiresponse_json)
multiresponse_json

[{'id': 'ABW',
  'iso2Code': 'AW',
  'name': 'Aruba',
  'region': {'id': 'LCN',
   'iso2code': 'ZJ',
   'value': 'Latin America & Caribbean '},
  'adminregion': {'id': '', 'iso2code': '', 'value': ''},
  'incomeLevel': {'id': 'HIC', 'iso2code': 'XD', 'value': 'High income'},
  'lendingType': {'id': 'LNX', 'iso2code': 'XX', 'value': 'Not classified'},
  'capitalCity': 'Oranjestad',
  'longitude': '-70.0167',
  'latitude': '12.5167'},
 {'id': 'AFE',
  'iso2Code': 'ZH',
  'name': 'Africa Eastern and Southern',
  'region': {'id': 'NA', 'iso2code': 'NA', 'value': 'Aggregates'},
  'adminregion': {'id': '', 'iso2code': '', 'value': ''},
  'incomeLevel': {'id': 'NA', 'iso2code': 'NA', 'value': 'Aggregates'},
  'lendingType': {'id': '', 'iso2code': '', 'value': 'Aggregates'},
  'capitalCity': '',
  'longitude': '',
  'latitude': ''},
 {'id': 'AFG',
  'iso2Code': 'AF',
  'name': 'Afghanistan',
  'region': {'id': 'MEA',
   'iso2code': 'ZQ',
   'value': 'Middle East, North Africa, Afghanistan & Pa

In [122]:
multiresponse_json[0:5]
#multiresponse_json[0]
#print(multiresponse_json[0]['name'],multiresponse_json[0]['incomeLevel']['value'] )

country_lst = []
inc_lvl_lst = []

for country in multiresponse_json:
    #print(country)
    #print(country['name'],country['incomeLevel']['value'] )
    country_lst.append(country['name'])
    inc_lvl_lst.append(country['incomeLevel']['value'])

countries_df = pd.DataFrame()
print(countries_df)
countries_df['country'] = country_lst
countries_df['incomeLevel'] = inc_lvl_lst
countries_df
#multiresponse_json[0]['incomeLevel']['value']
#print(multiresponse_json[0]['incomeLevel']['value'])

Empty DataFrame
Columns: []
Index: []


,country,incomeLevel
0,Aruba,High income
1,Africa Eastern and Southern,Aggregates
2,Afghanistan,Low income
3,Africa,Aggregates
4,Africa Western and Central,Aggregates
...,...,...
291,Sub-Saharan Africa excluding South Africa and ...,Aggregates
292,"Yemen, Rep.",Low income
293,South Africa,Upper middle income
294,Zambia,Lower middle income


In [127]:
country_dict = {'country' : ['C','B', 'C'],
                'income_lvl' : ['h', 'h', 'l']
                }
pd.DataFrame(country_dict)

,country,income_lvl
0,C,h
1,B,h
2,C,l


In [ ]:
country_lst = []
inc_lvl_lst = []

for country in multiresponse_json:
    #print(country)
    #print(country['name'],country['incomeLevel']['value'] )
    country_lst.append(country['name'])
    inc_lvl_lst.append(country['incomeLevel']['value'])

country_dict = {'country' : country_lst,
                'income_lvl' : inc_lvl_lst
                }

countries_df = pd.DataFrame(country_dict)
display(countries_df)


,country,income_lvl
0,Aruba,High income
1,Africa Eastern and Southern,Aggregates
2,Afghanistan,Low income
3,Africa,Aggregates
4,Africa Western and Central,Aggregates
...,...,...
291,Sub-Saharan Africa excluding South Africa and ...,Aggregates
292,"Yemen, Rep.",Low income
293,South Africa,Upper middle income
294,Zambia,Lower middle income


In [105]:
import pandas as pd
multiresponse_json[0:5]
#print(multiresponse_json[0]['name'])
#print(multiresponse_json[0]['incomeLevel']['value'])

country_lst = []
inc_lvl_lst = []
for country in multiresponse_json:
    #print(country['name'], country['incomeLevel']['value'])
    #print(country['incomeLevel']['value'])
    country_lst.append(country['name'])
    inc_lvl_lst.append(country['incomeLevel']['value'])

country_lst[0], inc_lvl_lst[0]

countries_df = pd.DataFrame()
countries_df['country'] = country_lst
countries_df['incomeLevel'] = inc_lvl_lst
countries_df

,country,incomeLevel
0,Aruba,High income
1,Africa Eastern and Southern,Aggregates
2,Afghanistan,Low income
3,Africa,Aggregates
4,Africa Western and Central,Aggregates
...,...,...
291,Sub-Saharan Africa excluding South Africa and ...,Aggregates
292,"Yemen, Rep.",Low income
293,South Africa,Upper middle income
294,Zambia,Lower middle income


__Exercise:__ From the information extracted, create the following dataframe: 

In [ ]:
##Your code here:


,country,income_lvl
0,Aruba,High income
1,Africa Eastern and Southern,Aggregates
2,Afghanistan,Low income
3,Africa,Aggregates
4,Africa Western and Central,Aggregates
5,Angola,Lower middle income
6,Albania,Upper middle income
7,Andorra,High income
8,Arab World,Aggregates
9,United Arab Emirates,High income
